tests/translator/input/embedded_connectors_esm_depends_on.yaml

```diff
@@ -0,0 +1,28 @@
+Resources:
+  MyFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Runtime: nodejs16.x
+      Handler: index.handler
+      InlineCode: |
+        exports.handler = async (event) => {
+          console.log(event);
+        };
+
+  MyQueue:
+    Type: AWS::SQS::Queue
+    Connectors:
+      MyConnector:
+        Properties:
+          Destination:
+            Id: MyFunction
+          Permissions:
+          - Read
+          - Write
+
+  MyEventSourceMapping:
+    DependsOn: MyQueueMyConnectorPolicy
```

Comment:
Should also have a test with `DependsOn` on the connector logical ID (verbatim). We automatically expand to the generated resource logical IDs:

https://github.com/aws/serverless-application-model/blob/b4f9ea1b083469a304ae248e26f3b2afb3a8b6f7/samtranslator/model/sam_resources.py#L1895-L1896